modules/@apostrophecms/ai-helper-image/ui/apos/components/AposAiHelperGenerateImage.vue

```diff
@@ -0,0 +1,124 @@
+<template>
+  <AposModal
+    class="apos-ai-helper-generate-image"
+    :modal="modal"
+    modal-title="aposAiHelper:generateImage"
+    @inactive="modal.active = false"
+    @show-modal="modal.showModal = true"
+    @esc="close"
+    @no-modal="$emit('safe-close')"
+  >
+    <template #main>
+      <AposModalBody>
+        <template #bodyMain>
+          <form class="apos-ai-helper-form">
+            <textarea v-model="prompt" :placeholder="$t('aposAiHelper:placeholderText')" />
+            <AposButton
+              :disabled="!prompt.length"
+              @click.prevent="generate"
+              :label="$t('aposAiHelper:generateImage')"
+            />
+            <p v-if="error">
+              An error occurred.
+            </p>
+            <div class="apos-ai-helper-images">
+              <button
+                v-for="image in images"
+                :key="image.id"
+                class="apos-ai-helper-image"
+                @click.prevent="save(image.id)"
+              >
+                <img :src="image.url">
+              </button>
+            </div>
+          </form>
+        </template>
+      </AposModalBody>
+    </template>
+  </AposModal>
+</template>
+
+<script>
+export default {
+  emits: [ 'modal-result' ],
+  data() {
+    return {
+      modal: {
+        active: false,
+        type: 'slide',
+        origin: 'right',
+        showModal: false,
+        width: 'full'
+      },
+      error: false,
+      prompt: '',
+      images: []
+    };
+  },
+  async mounted() {
+    this.modal.active = true;
+  },
+  methods: {
+    close() {
+      this.modal.showModal = false;
+    },
+    async generate() {
```

Comment:
Heads up for the real implementation - I usually avoid using methods that conflict with any mixins (you know how I feel about mixins anyway). `generate` as far as I remember has a "special meaning" when it comes to fields.

Reply:
This is a single-purpose modal, and I don't think I have any mixins in it, but I will check